plugin/pkg/shoot/validator/admission.go

```diff
@@ -932,28 +933,42 @@ func validateLoadBalancerProviderConstraints(providers []garden.OpenStackLoadBal
 	return false, validValues
 }
 
-func getMachineImage(machineImages []garden.MachineImage) (*garden.MachineImage, error) {
+func getMachineImage(machineImages []garden.MachineImage) (*garden.ShootMachineImage, error) {
 	if len(machineImages) == 0 {
 		return nil, errors.New("the cloud profile does not contain any machine image - cannot create shoot cluster")
 	}
-	// If the shoot does not specify a machine image then we consider the first one specified in the cloud profile to be the
-	// default machine image and return it.
-	return &machineImages[0], nil
+	// If the shoot does not specify a machine image then we determine the latest machine image versions in the cloudProfile, pick the first entry in the map and consider that as the default image
+	latestMachineImages, err := helper.DetermineLatestMachineImageVersions(machineImages)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine latest machine image from cloud profile: %s", err.Error())
+	}
+	var defaultImage garden.ShootMachineImage
+	for imageName, latestVersion := range latestMachineImages {
```

Comment:
We should provide a stable default image (the first specified in the `machineImages`) which you can't do by iterating over a map since the iteration order is randomized, see https://blog.golang.org/go-maps-in-action --> _Iteration Order_.